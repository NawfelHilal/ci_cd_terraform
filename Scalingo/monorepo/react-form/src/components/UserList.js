import React from "react";

const UserList = ({ users }) => {
  if (!Array.isArray(users)) {
    return (
      <div>
        <h2>Liste des utilisateurs</h2>
        <ul></ul>
      </div>
    );
  }
  return (
    <div>
      <h2>Liste des utilisateurs</h2>
      <ul>
        {users.map((u, i) => (
          <li key={i}>
            {u.firstName} {u.lastName} ({u.email})
          </li>
        ))}
      </ul>
    </div>
  );
};

export default UserList;
