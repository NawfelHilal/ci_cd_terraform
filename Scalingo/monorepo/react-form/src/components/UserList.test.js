import React from "react";
import { render, screen } from "@testing-library/react";
import UserList from "./UserList";

describe("UserList", () => {
  it("affiche la liste vide si users est un tableau vide", () => {
    render(<UserList users={[]} />);
    expect(screen.getByText(/Liste des utilisateurs/i)).toBeInTheDocument();
    // Aucun <li> ne doit être présent
    expect(document.querySelectorAll("ul li").length).toBe(0);
  });

  it("affiche les utilisateurs si users est un tableau rempli", () => {
    const users = [
      { firstName: "Jean", lastName: "Dupont", email: "jean@ex.com" },
      { firstName: "Marie", lastName: "Curie", email: "marie@ex.com" },
    ];
    render(<UserList users={users} />);
    expect(screen.getByText(/Jean Dupont/i)).toBeInTheDocument();
    expect(screen.getByText(/jean@ex.com/i)).toBeInTheDocument();
    expect(screen.getByText(/Marie Curie/i)).toBeInTheDocument();
    expect(screen.getByText(/marie@ex.com/i)).toBeInTheDocument();
    // Il doit y avoir 2 <li>
    expect(document.querySelectorAll("ul li").length).toBe(2);
  });

  it("affiche la liste vide si users est undefined", () => {
    render(<UserList users={undefined} />);
    expect(screen.getByText(/Liste des utilisateurs/i)).toBeInTheDocument();
    expect(document.querySelectorAll("ul li").length).toBe(0);
  });

  it("affiche la liste vide si users est null", () => {
    render(<UserList users={null} />);
    expect(screen.getByText(/Liste des utilisateurs/i)).toBeInTheDocument();
    expect(document.querySelectorAll("ul li").length).toBe(0);
  });

  it("affiche la liste vide si users est une string", () => {
    render(<UserList users={"not an array"} />);
    expect(screen.getByText(/Liste des utilisateurs/i)).toBeInTheDocument();
    expect(document.querySelectorAll("ul li").length).toBe(0);
  });

  it("gère les utilisateurs avec des champs manquants", () => {
    const users = [
      { firstName: "Jean" },
      { lastName: "Curie" },
      { email: "test@ex.com" },
      {},
    ];
    render(<UserList users={users} />);
    // Les <li> doivent être présents même si incomplets
    expect(document.querySelectorAll("ul li").length).toBe(4);
  });

  it("correspond au snapshot", () => {
    const users = [
      { firstName: "Jean", lastName: "Dupont", email: "jean@ex.com" },
    ];
    const { asFragment } = render(<UserList users={users} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
