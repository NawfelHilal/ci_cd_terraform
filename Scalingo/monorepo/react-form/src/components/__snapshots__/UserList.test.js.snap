// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList correspond au snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Liste des utilisateurs
    </h2>
    <ul>
      <li>
        Jean Dupont (jean@ex.com)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
